src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml.cs

```diff
@@ -35,6 +40,224 @@ public ProjectView()
             _versions.ItemContainerStyle = style;
         }
 
+        private TextBox _textBox;
+
+        private TextBox TextBox
+        {
+            get
+            {
+                if (_textBox == null) _textBox = _versions.Template.FindName("PART_EditableTextBox", _versions) as TextBox;
+                return _textBox;
+            }
+            set
+            {
+                _textBox = value;
+            }
+        }
+
+        private PackageDetailControlModel _packageDetailControlModel;
+
+        private PackageDetailControlModel PackageDetailControlModel
+        {
+            get
+            {
+                if (_packageDetailControlModel == null) _packageDetailControlModel = (PackageDetailControlModel)DataContext;
+                return _packageDetailControlModel;
+            }
+            set
+            {
+                _packageDetailControlModel = value;
+            }
+        }
+
+        private DetailControlModel _detailModel;
+
+        private DetailControlModel DetailModel
+        {
+            get
+            {
+                if (_detailModel == null) _detailModel = (DetailControlModel)DataContext;
+                return _detailModel;
+            }
+            set
+            {
+                _detailModel = value;
+            }
+        }
+
+        private string _previousFilter;
+        private string PreviousText
+        {
+            get
+            {
+                if (_previousFilter != null) return _previousFilter;
+                return string.Empty;
+            }
+            set
+            {
+                _previousFilter = value;
+            }
+        }
+
+        protected override void OnPreviewKeyDown(KeyEventArgs e)
```

Comment:
What happens if I copy/paste text into the field.

like (1.0,2.0) from https://docs.microsoft.com/en-us/nuget/concepts/package-versioning#version-ranges